qa-pipelines/qa-pipeline.yml.erb

```diff
@@ -227,26 +255,38 @@ jobs:
     # We intentionally don't put the teardown and pool release steps in an ensure
     # block, so that when tests fail we have a chance of examining why things are
     # failing.
-    <% if enable_cf_teardown %>
+    <% if enable_cf_teardown && !enable_terraform %>
     - task: cf-teardown
       file: ci/qa-pipelines/tasks/cf-teardown.yml
       timeout: 1h
       params: *common-task-params
-    <% end %> # enable_cf_teardown
+    <% end %> # enable_cf_teardown and not enable_terraform
+    <% if !enable_terraform %>
     - put: pool.kube-hosts
       params: {release: pool.kube-hosts}
+    <% end %> # not enable_terraform
     <% if status_reporting %>
     on_success:
       put: status.src
       params:
-        <<: *status-params-<%= avail %>
+        << : *status-params-<%= avail %>
         state: success
     <% end %> # status_reporting
     <% if status_reporting %>
     on_failure:
       put: status.src
       params:
-        <<: *status-params-<%= avail %>
+        << : *status-params-<%= avail %>
         state: failure
     <% end %> # status_reporting
+    <% if enable_terraform %>
```

Comment:
Shouldn't terraform teardown task templating be informed by the `enable-teardown` flag also?

Reply:
no. The terraform CI work follow this guideline:  "QA CI don't care if user want to enable-teardown or not. CI will tear it down."